app/models/bet.rb

```diff
@@ -2,4 +2,42 @@ class Bet < ApplicationRecord
   has_many :users, through: :user_bets
   has_many :user_bets
   has_many :options, through: :user_bets, source: :bet_option
+
+  def resolve
+    losers.each do |debtor|
+      debt = UserBet.find_by(user: debtor, bet: self).amount_bet
+      win_contributions.each do |contribution|
+        Debt.create(debtor: debtor,
+                    creditor: contribution[:creditor],
+                    amount:   debt * contribution[:percentage])
+      end
+    end
+  end
+
+  def win_option
```

Comment:
Should be a scope on Option

http://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope